API proxy pro Nedlužím státu

zoul opened the pull request:
Zatím jen první nástřel, [ukázkově nasazeno tady](https://cd-tools-git-feature-nedluzim-statu-ceskodigital.vercel.app/api/nedluzimstatu/spadovost). FYI @stefanides.

zoul commented on api/nedluzimstatu/_shared.ts:
```diff
@@ -0,0 +1,34 @@
+  request: express.Request,
+  _: express.Response
+) {
+  let origin = request.headers.origin;
```
Tohle může být podle překladače `string | string[] | undefined`, tak je potřeba se s tím korektně vypořádat?

stefanides replied:
@zoul opraveno. netuším, co může být v tom poli - ale když bude, použije se první hodnota.

zoul commented on api/nedluzimstatu/_shared.ts:
```diff
@@ -0,0 +1,34 @@
+  proxyResponse.headers["Cache-Control"] = httpCacheControl;
+}
+
+export function runMiddleware(req: any, res: any, fn: any) {
```
Tady netuším správné typy, `any` je tam čistě jako vztyčený prostředníček překladači. Víte někdo typy?

zoul replied:
Shodli jsme se na tom, že na to kašlem.

zoul commented on api/nedluzimstatu/ovm.ts:
```diff
@@ -0,0 +1,34 @@
+const proxyTimeout = 8000;
+
+const ovmApi = createProxyMiddleware({
+  target: "https://api.apitalks.store/ovm",
```
Tohle je 404, bude potřeba najít správný endpoint.

stefanides replied:
endpoint je dobře, viz (https://www.api.store/organy-verejne-moci/), ale úplně jsem nepochopil mapování cesty z requestu na proxyrequest (i bez vercelu se mi tam jednou kus původní cesty zkopíruje a podruhé ne). veřejná cesta k api musí nabízet obojí, např. to /ovm a /spadovost

zoul replied:
A je nutný používat ten proxy middleware? Nejde prostě přebrat parametry od klienta, předat je „ručně“ ApiTalks a výsledek zase předat zpátky klientovi?

stefanides replied:
@zoul úplně nutné to není, pokud něco takového umí vercel, můžeme použít. Přepisovat hotovou proxy knihovnu mi ale moc smysl nedává. (dělají to takhle i jinde, např. https://github.com/vercel/next.js/discussions/14057)

stefanides replied:
@zoul podařilo se k tomu něco najít? v kódu mi nějak chybí definice zdrojové URI, která se dá přes http volat - kde se nastavuje? aneb neměl by být nějaký další rewrite i v now.json?

zoul reviewed (commented):
Nechal jsem pár poznámek k tomu, co je potřeba dodělat. Jak to bude, můžem někoho poprosit o pořádné review.

stefanides committed:
Added proxy for api.store API used in nedluzimstatu.cz:

Add first working draft

Extract shared code to a standalone file

Add OVM endpoint support

Move all Nedlužím státu code to /nedluzimstatu

nedluzimstatu/_shared.ts: fix type check

readme.md: update with api.store proxy

nedluzimstatu/ovm.ts: fix target url

config: rewrite for nedluzimstatu/ovm

nedluzimstatu/ovm.ts: fix empty id

zoul commented:
@miiila, neměl bys prosím chvilku na závěrečné review? Díky!
